### PR TITLE
WIP: Add cspr and tcspr to core module

### PR DESCRIPTION
**Pull request by hgonzalez-simtlix**
Jira card: https://simtlix.atlassian.net/browse/BGA-1165

### CODE REVIEW

File: modules/core/src/v2/coins/tcspr.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { BaseCoin } from '../baseCoin';
+import { Cspr } from './cspr';
+
+export class Tcspr extends Cspr {
+  static createInstance(bitgo: BitGo): BaseCoin {
+    return new Tcspr(bitgo);
+  }
+
+  getChain() {
+    return 'teth';
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/tcspr.ts:11]: Change this for tcspr

File: modules/core/src/v2/coins/tcspr.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+  getFullName() {
+    return 'Testnet Ethereum';
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/tcspr.ts:15]: Tesnet Casper

**Review by rliberal-simtlix — CHANGES_REQUESTED** [review]

File: modules/core/src/v2/coins/cspr.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+export class Cspr extends BaseCoin {
+    
+    protected readonly _staticsCoin: Readonly<StaticsBaseCoin>;
+
+    constructor(bitgo: BitGo, staticsCoin?: Readonly<StaticsBaseCoin>) {
+        super(bitgo);
+    
+        if (!staticsCoin) {
+          throw new Error('missing required constructor parameter staticsCoin');
+        }
```
Comment (FabianSimtlix) [inline modules/core/src/v2/coins/cspr.ts:18]: Why is the parameter optional if we are going to throw an exception when it's missing?

File: modules/core/src/v2/coins/cspr.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    /**
+   * Generate BLS key pair
+   *
+   * @param seed - byte array to generate BLS key pair from
+   * @returns {Object} object with generated pub and prv
+   */
+    generateKeyPair(seed?: Buffer): KeyPair {
+    const keyPair = seed ? new CsprAccountLib.KeyPair({ seed }) : 
+    new CsprAccountLib.KeyPair();
```
Comment (FabianSimtlix) [inline modules/core/src/v2/coins/cspr.ts:57]: Indentation. It's not clear that this one correspondes with previous assignment

File: modules/core/src/v2/coins/tcspr.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+  getChain() {
+    return 'tcspr';
+  }
+
+  getFullName() {
+    return 'Testnet Casper';
+  }
+}
```
Comment (FabianSimtlix) [inline modules/core/src/v2/coins/tcspr.ts:17]: This values should be taken from statics

**Review by FabianSimtlix — CHANGES_REQUESTED** [review]

File: modules/core/test/v2/unit/coins/cspr.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+  it('should instantiate the coin', function () {
+    let localBasecoin = bitgo.coin('tcspr');
+    localBasecoin.should.be.an.instanceof(Tcspr);
+
+    localBasecoin = bitgo.coin('cspr');
+    localBasecoin.should.be.an.instanceof(Cspr);
+  });
+
+  it('should return cspr', function () {
```
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/coins/cspr.ts:25]: should return tcspr?

**Review by rliberal-simtlix — APPROVED** [review]

File: modules/core/src/v2/coins/cspr.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+import { BaseCoin as StaticsBaseCoin } from '@bitgo/statics';
+
+import Bluebird = require('bluebird');
+
+export class Cspr extends BaseCoin {
+    
+    protected readonly _staticsCoin: Readonly<StaticsBaseCoin>;
+
+    constructor(bitgo: BitGo, staticsCoin?: Readonly<StaticsBaseCoin>) {
+      super(bitgo);
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/cspr.ts:14]: It's a bit odd to alternate between 4 and 2 space indent. Not a blocker though. You can add the following comment at the top of the file to have prettier auto-format any changes so you don't need to think about this:
```
/*
 * @prettier
 */
```

**Review by tylerlevine — DISMISSED** [review]
lgtm, just one nit on the white space but it's not a blocker.

**Review by rliberal-simtlix — APPROVED** [review]

**Review by CapnMigraine — DISMISSED** [review]
LGTM. Please squash your commits.

**Review by CapnMigraine — DISMISSED** [review]
lgtm

File: modules/core/src/v2/coins/index.ts [hunk 8]
```diff
@@ -47,3 +47,5 @@ export { Tzec } from './tzec';
 export { Xlm } from './xlm';
 export { Xrp } from './xrp';
 export { Zec } from './zec';
+export { Cspr } from './cspr';
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/index.ts:50]: Please maintain alphabetic ordering

**Review by tylerlevine — APPROVED** [review]

**Review by CapnMigraine — APPROVED** [review]

**Pull request merged**